workers/unity/Packages/com.improbable.gdk.playerlifecycle/PlayerLifecycleHelper.cs

```diff
@@ -43,24 +45,14 @@ public static bool SerializeArguments(Object playerCreationArguments, out byte[]
             }
         }
 
-        public static bool DeserializeArguments<T>(byte[] serializedArguments, out T deserializedArguments)
+        public static T DeserializeArguments<T>(byte[] serializedArguments)
         {
-            try
-            {
-                using (var memoryStream = new MemoryStream())
-                {
-                    var binaryFormatter = new BinaryFormatter();
-                    memoryStream.Write(serializedArguments, 0, serializedArguments.Length);
-                    memoryStream.Seek(0, SeekOrigin.Begin);
-                    deserializedArguments = (T) binaryFormatter.Deserialize(memoryStream);
-                    return true;
-                }
-            }
-            catch (Exception e)
+            using (var memoryStream = new MemoryStream())
             {
-                UnityEngine.Debug.LogError($"Unable to deserialize player creation arguments. {e.Message}");
-                deserializedArguments = default;
-                return false;
+                var binaryFormatter = new BinaryFormatter();
```

Comment:
There's now asymmetry between this and the Serialize. One throws, one catches the exception and logs

Reply:
ok, so now both will throw and it's up to the user to handle these exceptions (probably for the better as caller has context)